include/object/domain.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2025, Indan Zupancic
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#pragma once
+
+#include <config.h>
+#include <types.h>
+#include <api/failures.h>
+
+/* The domain is stored in the top 8 bits, duration in the bottom 56 bits.
+ * Not using 64 bits for duration also avoids overflows. */
+typedef uint64_t dschedule_t;
+
+#define DSCHED_END_MARKER 0
+#define DOMAIN_0_MAX_DURATION (UINT64_MAX >> 8)
+
+static inline dom_t dschedule_domain(dschedule_t sched)
+{
+    return sched >> 56;
+}
+
+static inline ticks_t dschedule_duration(dschedule_t sched)
+{
+    return sched & DOMAIN_0_MAX_DURATION;
```

Comment:
feels like calling this something like `DOMAIN_DURATION_MASK` or something makes more sense; it's coincidental that the duration and the mask happen to be the same, but really it is a mask

@lsf37 do we care that this is not using the bitfield generator? we've had some issues in the past with people manually doing word arithmetic and masking causing us hiccups, because the generator spits out useful proofs and the direct way doesn't